### ISSUE TITLE
Updating the Secret not working

### ISSUE THREAD

**Issue by manderson-it** [titansoft-pte-ltd/imagepullsecret-patcher#17]
Hey everyone,

How can you update the image-pull-secret? Let's say the Secret to pull images needs rotation after one year.
Would be useful to document the process in this repository.
Running Kubernetes v1.17.13

The documentation in this repository and this link [https://kubernetes.io/docs/concepts/configuration/secret/#mounted-secrets-are-updated-automatically](https://kubernetes.io/docs/concepts/configuration/secret/#mounted-secrets-are-updated-automatically) suggest that the key to the container registry can be updated dynamically.

I am using the `v0.14` release with those original manifests files. I commented out the Secret and added the Secret myself with this command:
```
kubectl create secret docker-registry image-pull-secret \
-n imagepullsecret-patcher \
--docker-server=gcr.io \
--docker-username=_json_key \
--docker-password="$(cat ~/gcr-test-59358d4f6058.json)" \
--docker-email=any@valid.email
```

The Secret is correctly patched to other namespaces and it allows to pull images, as desired.

**Updating that Secret is not working**
Neither of these methods work. Every method outlined below results in the old `.dockerconfigjson` value.
The commands below work syntactically, but the old value for `.dockerconfigjson` is still present instead of the new one from the new SA JSON key file. I am checking the value with
`kubectl get secrets -n imagepullsecret-patcher image-pull-secret -o yaml`

Seems like the Deployment is overwriting it?

```
# Try to edit it
kubectl -n imagepullsecret-patcher edit secret image-pull-secret
secret/image-pull-secret edited

# Delete the Secret and immediately create the Secret with a new SA JSON key file
kubectl delete secrets image-pull-secret -n imagepullsecret-patcher && kubectl create secret docker-registry image-pull-secret -n imagepullsecret-patcher --docker-server=gcr.io --docker-username=_json_key --docker-password="$(cat ~/Downloads/marek-gcr-test-66658a4a1234.json)" --docker-email=any@valid.email
secret "image-pull-secret" deleted
secret/image-pull-secret created

# Try with a piped replace operation
kubectl create secret docker-registry image-pull-secret \
-n imagepullsecret-patcher \
--docker-server=gcr.io \
--docker-username=_json_key \
--docker-password="$(cat ~/gcr-test-66658a4a1234.json)" \
--docker-email=any@valid.email \
--dry-run -o yaml | kubectl replace --force -f -

# Try removing the annotation managed-by
# annotation gets re-added immediately
kubectl annotate secret image-pull-secret app.kubernetes.io/managed-by-
secret/image-pull-secret annotated

# Prepare a patch YAML file with the new base64 value of the new SA JSON key file
kubectl -n imagepullsecret-patcher patch secret image-pull-secret --patch "$(cat patch-file.yaml)"
secret/image-pull-secret patched
# The old value of .dockerconfigjson is showing
```

**Comment by js-timbirkett** [comment]
I think I see the issue...

In the example deployment, the secret that is mounted as the docker config to be propagated is called `image-pull-secret` and the secret that is propagated is called `image-pull-secret` this probably creates some sort of cached circular dependency or something...

I'd try changing the source secret to something else like:
```
apiVersion: v1
kind: Secret
type: kubernetes.io/dockerconfigjson
metadata:
  name: image-pull-secret-src
  namespace: imagepullsecret-patcher
....
....
```

and change the deployment to mount that secret:
```
....
....

      volumes:
        - name: src-dockerconfigjson
          secret: 
            secretName: image-pull-secret-src
```

that should decouple the source secret from the secrets that get patched and prevent issues with updating the secret.

**Comment by manderson-it** [comment]
Nice! Different names resolve the rotation issue I experienced. Will create a PR shortly @js-timbirkett

**Comment by manderson-it** [comment]
@js-timbirkett provided a PR, see above. Thanks a lot for your time!

**Comment by billjh** [comment]
Should be resolved by PR #18
